src/helpers/jsonValidation.ts

```diff
@@ -0,0 +1,63 @@
+import _isArray from 'lodash/isArray';
+import _isObjectLike from 'lodash/isObjectLike';
+import { ConstructionSite } from '../types';
+
+const isStringOrUndefined = (value: unknown): value is string | undefined => {
+  return typeof value === 'string' || value === undefined;
+};
+
+const isGeoLocationOrUndefined = (
+  value: unknown
+): value is { lat: number; lon: number } | undefined => {
+  if (value === undefined) return true;
+  return (
+    typeof _isObjectLike(value) &&
+    typeof (value as { lat: unknown }).lat === 'number' &&
+    typeof (value as { lon: unknown }).lon === 'number'
+  );
+};
+
+const isStringArrayOrUndefined = (value: unknown): value is string[] | undefined => {
+  if (value === undefined) return true;
+  if (_isArray(value))
+    return value.reduce((accumulated, current) => accumulated && typeof current === 'string', true);
+  return false;
+};
+
+const checkConstructionSiteTypes = (item: unknown) => {
+  const {
+    category,
+    cause,
+    description,
+    direction,
+    endDate,
+    imageUri,
+    location,
+    locationDescription,
+    restrictions,
+    startDate,
+    title
+  } = item as ConstructionSite;
+
+  return (
+    typeof title === 'string' &&
+    typeof startDate === 'string' &&
+    isStringOrUndefined(category) &&
+    isStringOrUndefined(description) &&
+    isStringOrUndefined(cause) &&
+    isStringOrUndefined(direction) &&
+    isStringOrUndefined(endDate) &&
+    isStringOrUndefined(imageUri) &&
+    isStringOrUndefined(locationDescription) &&
+    isStringArrayOrUndefined(restrictions) &&
+    isGeoLocationOrUndefined(location)
+  );
+};
+
+export const filterForValidConstructionSites = (json: unknown): ConstructionSite[] => {
+  if (!_isArray(json)) return [];
+  return json.filter((item) => {
+    if (!item) return false;
+    return checkConstructionSiteTypes(item);
+  });
```

Comment:
there are several places in this file with close `return`s to each other, which is heavy to read. let us plan a call with @SoleCincis in order to find a way to improve the readability in this file?!

Reply:
meeting is scheduled 💪

Reply:
we can refactor it at a later date after the meeting. i suggest merging it as is for now

Reply:
ok, i linked this discussion in my calendar event for tomorrow.